### PR TITLE
Add serialization API to brute-force

### PR DESCRIPTION
**Pull request by lowener**
I noticed it was missing while switching Milvus to cuVS

### CODE REVIEW

File: python/cuvs/cuvs/neighbors/brute_force/brute_force.pyx [hunk 1]
```diff
@@ -256,3 +257,73 @@ def search(Index index,
         ))
 
     return (distances, neighbors)
+
+
+@auto_sync_resources
+def save(filename, Index index, bool include_dataset=True, resources=None):
```
Comment (cjnolet) [inline python/cuvs/cuvs/neighbors/brute_force/brute_force.pyx:263]: Can you please add the Python and C++ brute force serialization APIs to the docs?

File: cpp/include/cuvs/neighbors/brute_force.h [hunk 2]
```diff
@@ -166,6 +166,51 @@ cuvsError_t cuvsBruteForceSearch(cuvsResources_t res,
...
+
+/**
+ * Load index from file.
+ *
+ * Experimental, both the API and the serialization format are subject to change.
+ *
+ * @param[in] res cuvsResources_t opaque C handle
+ * @param[in] filename the name of the file that stores the index
+ * @param[out] index BRUTEFORCE index loaded disk
+ */
```
Comment (cjnolet) [inline cpp/include/cuvs/neighbors/brute_force.h:206]: Please provide a quick usage example here too.

File: cpp/include/cuvs/neighbors/brute_force.hpp [hunk 3]
```diff
@@ -375,4 +383,230 @@ void search(raft::resources const& handle,
...
  */
 
+/**
+ * @defgroup bruteforce_cpp_index_serialize Bruteforce index serialize functions
+ * @{
+ */
+/**
+ * Save the index to file.
+ *
+ * Experimental, both the API and the serialization format are subject to change.
```
Comment (cjnolet) [inline cpp/include/cuvs/neighbors/brute_force.hpp:393]: We may be able to remove this now (everywhere). This API has been around for awhile in RAFT and in the other indexes.

File: python/cuvs/cuvs/neighbors/brute_force/brute_force.pyx [hunk 4]
```diff
@@ -256,3 +257,73 @@ def search(Index index,
...
+    """
+    cdef string c_filename = filename.encode('utf-8')
+    cdef cuvsResources_t res = <cuvsResources_t>resources.get_c_obj()
+    check_cuvs(cuvsBruteForceSerialize(res,
+                                       c_filename.c_str(),
+                                       index.index))
+
+
+@auto_sync_resources
+def load(filename, resources=None):
```
Comment (cjnolet) [inline python/cuvs/cuvs/neighbors/brute_force/brute_force.pyx:300]: Please provide a usage example here as well.

File: cpp/include/cuvs/neighbors/brute_force.hpp (hunk 3 above)
Comment (lowener) [reply cpp/include/cuvs/neighbors/brute_force.hpp]: I removed the mention of experimental, but I still kept the mention that the serialization format can be subject to changes to not have users expect retro-compatibility.

**Review by cjnolet — APPROVED** [review]

**Pull request merged**